client/homebrew/pages/userPage/userPage.jsx

```diff
@@ -14,69 +14,50 @@ const HelpNavItem = require('../../navbar/help.navitem.jsx');
 const ErrorNavItem = require('../../navbar/error-navitem.jsx');
 const VaultNavitem = require('../../navbar/vault.navitem.jsx');
 
-const UserPage = createClass({
-	displayName     : 'UserPage',
-	getDefaultProps : function() {
-		return {
-			username : '',
-			brews    : [],
-			query    : '',
-			error    : null
-		};
-	},
-	getInitialState : function() {
-		const usernameWithS = this.props.username + (this.props.username.endsWith('s') ? `’` : `’s`);
-
-		const brews = _.groupBy(this.props.brews, (brew)=>{
-			return (brew.published ? 'published' : 'private');
-		});
-
-		const brewCollection = [
-			{
-				title : `${usernameWithS} published brews`,
-				class : 'published',
-				brews : brews.published
-			}
-		];
-		if(this.props.username == global.account?.username){
-			brewCollection.push(
-				{
-					title : `${usernameWithS} unpublished brews`,
-					class : 'unpublished',
-					brews : brews.private
-				}
-			);
-		}
-
-		return {
-			brewCollection : brewCollection
-		};
-	},
-	errorReported : function(error) {
-		this.setState({
-			error
-		});
-	},
-
-	navItems : function() {
-		return <Navbar>
-			<Nav.section>
-				{this.state.error ?
-					<ErrorNavItem error={this.state.error} parent={this}></ErrorNavItem> :
-					null
-				}
-				<NewBrew />
-				<HelpNavItem />
-				<VaultNavitem/>
-				<RecentNavItem />
-				<Account />
-			</Nav.section>
-		</Navbar>;
-	},
-
-	render : function(){
-		return <ListPage brewCollection={this.state.brewCollection} navItems={this.navItems()} query={this.props.query} reportError={this.errorReported}></ListPage>;
-	}
-});
+const UserPage = (props) => {
+    const { username = '', brews = [], query = '', error = null } = props;
+    const usernameWithS = username + (username.endsWith('s') ? `’` : `’s`);
+
+    const groupedBrews = _.groupBy(brews, (brew) => {
+        return brew.published ? 'published' : 'private';
+    });
+
+	const brewCollection = [
+		{
+			title: `${usernameWithS} published brews`,
+			class: 'published',
+			brews: groupedBrews.published || []
+		},
+		...(username === global.account?.username ? [{
+			title: `${usernameWithS} unpublished brews`,
+			class: 'unpublished',
+			brews: groupedBrews.private || []
+		}] : [])
+	];	
+
+    const [currentError, setCurrentError] = useState(error);
```

Comment:
My preference would be to have the `state` stuff placed right at the top of the component below props, before any other functions since that is traditionally where we have placed it.  And it's nice to see what variables are *props* and which are *state* right at the beginning.